### DISCUSSION TITLE
How to configure keypad module with remore relays modules

### DISCUSSION THREAD

**Discussion by zakdma** [arendst/Tasmota#22585] [General]
Hi!

I need to create such a configuration.
One module with keypad (5 switches)
Every switch controls the separate remote relay module.
When switch1 is closed then remote relay1 is ON.
When switch2 is closed then remote relay2 is ON.
And so on.
I can achieve this for one link Switch1/Relay1 by using the same GroupName for the keypad module channel 1 and remote Relay1.
But it needs a keypad module local Relay1 pin to be also configured. In this case link Switch1->Relay1 is working fine.
But there are problems
1. There can be only max 4 channels GroupName specified. But I need 5.
2. There are not enough free pins (wemos d1 mini) to configure 5 pairs of Switch/Relay pins.

What is the best approach to achieve the configuration I described? Is it even possible?
Please guide me in the right direction.

Thanks!

**Comment by joba-1** [comment]
my first impuls would be using rules. trigger on switch state and action sends mqtt topic of remote relay

**Reply by zakdma** [reply]
@joba-1 thank you. Is there any example of such config (rule)?
Also, there is no MQTT broker in the local network.

**Comment by joba-1** [comment]
Not an expert on rules, but here is the documentation: https://tasmota.github.io/docs/Rules

With a quick look at the page I guess you probably need something like 

`rule1 ON switch1#state DO publish cmnd/[your-remoterelay1-topic]/POWER %value% ENDON ON switch2#state DO publish cmnd/[your-remoterelay2-topic]/POWER %value ENDON ...`

Probably also need to define a set of rules once connected (ON mqtt#connected DO ...) to set the initial state

If you don't have a broker (bad choice...) probably some kind of websend [ip/hostname of remote] power1 %value% instead of publish and wifi#connected for mqtt#connected

Finally enable the rule1 with rule1 1

All above untested and taken from above docs. Lots of examples there with exact syntax...

**Comment by zakdma** [comment]
I finally resolved this problem. I created 5 switches on different pins and created rules
```
Rule1 
ON switch1#state DO WebSend [192.168.0.160] POWER %value% ENDON
ON switch2#state DO WebSend [192.168.0.161] POWER %value% ENDON
ON switch3#state DO WebSend [192.168.0.162] POWER %value% ENDON
ON switch4#state DO WebSend [192.168.0.163] POWER %value% ENDON
ON switch5#state DO WebSend [192.168.0.164] POWER %value% ENDON
BREAK
```

All is working fine and smoothly.

I stick to the principle that such simple controls should work without any need for additional infrastructure like MQTT Broker, HA etc.
So, for instance, lights should work even if my server dies.
Additional infrastructure should extend the existing setup but not replace it.

Thanks for the help!

**Reply by barbudor** [reply]
An alternative might be to create virtual relays on your keypad devices and use DeviceGroups to sync them. Sync happens in the background using UDP messages

**Reply by sfromis** [reply]
I think that the Device Groups feature was already used, but with only 4 groups it did not work out for 5 relay devices. And it would need 5 free pins on the single keypad device, which can be a stretch on ESP8266.

**Reply by barbudor** [reply]
You can also create more virtual realy with only 3 pins by using the HC595 driver
No need to have a HC595 and you can add up to 32 virtual relays

Regarding devicegroups the limit is actually 4 groups but each can have multiple relays (up to 32 ?)
So by tweaking the relay indexes on the receiving devices you could be able to achieve the same wiht only 1 group
But I haven't tried
